okio/src/commonMain/kotlin/okio/-Util.kt

```diff
@@ -112,5 +113,18 @@ internal fun Int.toHexString(): String {
   result[7] = HEX_DIGITS[this shr 8  and 0xf] // ktlint-disable no-multi-spaces
   result[8] = HEX_DIGITS[this shr 4  and 0xf] // ktlint-disable no-multi-spaces
   result[9] = HEX_DIGITS[this        and 0xf] // ktlint-disable no-multi-spaces
-  return String(result)
+
+  // Find the first non-zero index
+  var i = 2
+  while (i < 10) {
+    if (result[i] != '0') break
+    i++
+  }
+
+  // truncate on a whole byte and shift to start of "0x"
+  i -= 2 + (i % 2)
```

Comment:
So this does something a little fancy: it only truncates the string on the byte boundary. I was just kind of having fun with this so if you just want to truncate on the first non-zero character, it's really easy to change.

For example
```kotlin
fun main() {
  println(0x12345678.toHexString())
  println(0x1234567.toHexString())
  println(0x123456.toHexString())
  println(0x12345.toHexString())
  println(0x1234.toHexString())
  println(0x123.toHexString())
  println(0x12.toHexString())
  println(0x1.toHexString())
  println(0x0.toHexString())
}
```
will produce
```
0x12345678
0x01234567
0x123456
0x012345
0x1234
0x0123
0x12
0x01
0x00
```

Reply:
I think this is cute and clever and awesome. But I don’t think we should ship it ’cause an Int isn’t a `byte[]`, even though it serves that role well.

Reply:
No problem. It was late, I was having fun. 😃